src/components/File/hooks/use-file-by-filepath.ts

```diff
@@ -68,24 +73,29 @@ export const useFileByFilepath = () => {
     filePath: string | null,
     indexFileInDatabase: boolean = false
   ) => {
-    if (editor?.getHTML() !== null && filePath !== null) {
-      const markdown = editor?.storage.markdown.getMarkdown();
+    const markdownContent = editor?.storage.markdown.getMarkdown();
+    if (
+      markdownContent !== null &&
+      filePath !== null &&
+      isFileContentModified
+    ) {
       await window.files.writeFile({
         filePath: filePath,
-        content: markdown,
+        content: markdownContent,
       });
+
+      setIsFileContentModified(false);
+
       if (indexFileInDatabase) {
-        await window.files.indexFileInDatabase(filePath);
+        window.files.indexFileInDatabase(filePath);
       }
     }
   };
-  // read file, load content into fileContent
-  const openFileByPath = async (newFilePath: string) => {
-    //if the fileContent is null or if there is no file currently selected
 
+  const openFileByPath = async (newFilePath: string) => {
     saveEditorContentToPath(editor, currentlyOpenedFilePath, true);
-
     const fileContent = (await window.files.readFile(newFilePath)) ?? "";
+    setIsFileContentModified(false);
```

Comment:
saveEditorContentToPath already sets isFileContentModified to false, I think

Reply:
You're right it's fixed